kolibri/plugins/slideshow_render/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -0,0 +1,214 @@
+        type: Object,
+        default: () => {},
+      },
+      files: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data: () => ({
+      isInFullscreen: false,
+      manifest: {},
+      slides: null,
+      currentSlide: null,
+      hooperSettings: {
+        centerMode: true,
+      },
+    }),
+    computed: {
+      slideshow_images: function() {
+        const files = this.files;
+        return files.filter(file => file.preset != 'Slideshow Manifest');
+      },
+      caption_height: function() {
+        return 30 + (this.currentSlide ? this.$refs[this.currentSlide.id][0].clientHeight : 0);
+      },
+    },
+    mounted() {
+      /*
+        Using the manifest file, get the JSON from the manifest, then
+        use the manifest JSON to get all slide images and metadata.
+      */
+      fetch(this.defaultFile.storage_url, { method: 'GET' })
+        .then(response => {
+          return response.status === 200 ? response.json() : {};
+        })
+        .then(manifest_data => {
+          this.manifest = manifest_data.slideshow_data;
+
+          this.slides = _.orderBy(
+            this.manifest.map(image => {
+              return {
+                storage_url: this.slideshow_images.find(
+                  sFile =>
+                    sFile.storage_url
+                      .split('/')
+                      [sFile.storage_url.split('/').length - 1].split('.')[0] == image.checksum
+                ).storage_url,
+                caption: image.caption,
+                sort_order: image.sort_order,
+                id: image.checksum,
+                descriptive_text: image.descriptive_text,
+              };
+            }),
+            ['sort_order'],
+            ['asc']
+          );
+        });
+    },
+    methods: {
+      handleSlide(payload) {
+        this.currentSlide = this.slides[payload.currentSlide];
+      },
+      slideTextId(id) {
+        return 'descriptive-text-' + id;
+      },
+    },
+    $trs: {
+      exitFullscreen: 'Exit fullscreen',
+      enterFullscreen: 'Enter fullscreen',
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .btn {
+    position: absolute;
+    top: 8px;
+    right: 8px;
+    z-index: 100;
+    fill: white;
+  }
+  .slideshow-renderer {
+    position: relative;
+    height: 70vh;
+    overflow-x: auto;
+    overflow-y: hidden;
+    text-align: center;
+  }
+  .hidden-descriptive-text {
+    position: absolute;
+    top: auto;
+    left: -10000px;
+    width: 1px;
+    height: 1px;
+    overflow: hidden;
+  }
+  .slideshow-slide-image {
+    position: relative;
+    box-sizing: content-box;
+    width: calc(100% - 100px);
+    height: calc(
+      100% - 50px
+    ); // Very much not going to be this way once the caption positioning is figured out
+
+    margin: 0 auto;
+    background-repeat: no-repeat;
+    background-position: center center;
+    background-size: contain;
+
+    img {
+      object-fit: contain;
```

Comment:
If you don't already, I think you'll need a polyfill for this. IE 11 and old Android browsers do not support `object-fit`

Reply:
Thanks @bjester - I hadn't considered how new object-fit is - I found and implemented a simple and working polyfill. In testing I was also was reminded that IE11 doesn't support fetch() so had to revise that to use XMLHTTPRequest.

Reply:
Nice! Yeah, object-fit is really nice so glad the polyfill worked!